test/Sentry.Tests/Internals/FileSystemTests.cs

```diff
@@ -0,0 +1,56 @@
+using System.IO.Abstractions.TestingHelpers;
+
+namespace Sentry.Tests.Internals;
+
+public class FileSystemTests
+{
+    private const string Filename = "lock.txt";
+
+    [Fact]
+    public void Leased_file_cannot_be_leased_again()
+    {
+        const string path = @$"C:\test\{Filename}";
+
+        IDictionary<string, MockFileData> files  = new Dictionary<string, MockFileData>
+        {
+            { path, new MockFileData(Filename) { AllowedFileShare = FileShare.None } }
+        };
+
+        IFileSystem fileSystem = new FakeFileSystem(files);
+
+        Assert.Throws<IOException>(() =>
+        {
+            using (fileSystem.GetLeaseFile(path)) { }
```

Comment:
I don't understand what we're testing here if this file system is getting mocked. Could you please expand?

Reply:
The direction was originally to verify this against TestFileSystem to avoid what was described as issues in the past.
Sorry, I don't have the full context. I did look into this, and this test was an attempt to see if it's feasible, but it may not help as the OS is the one that releases a file, and a real OS would be needed.